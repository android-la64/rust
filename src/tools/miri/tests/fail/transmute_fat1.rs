// error-pattern: type validation failed: encountered a pointer
// normalize-stderr-test: "\[u8; (08|16)\]" -> "$$ARRAY"

fn main() {
    #[cfg(target_pointer_width = "64")]
    let bad = unsafe { std::mem::transmute::<&[u8], [u8; 16]>(&[1u8]) };
    #[cfg(target_pointer_width = "32")]
    let bad = unsafe { std::mem::transmute::<&[u8], [u8; 08]>(&[1u8]) };
    let _val = bad[0] + bad[bad.len() - 1];
}
