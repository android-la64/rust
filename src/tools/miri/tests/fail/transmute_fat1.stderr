error: Undefined Behavior: type validation failed: encountered a pointer, but expected plain (non-pointer) bytes
  --> $DIR/transmute_fat1.rs:LL:CC
   |
LL |     let bad = unsafe { std::mem::transmute::<&[u8], $ARRAY>(&[1u8]) };
   |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ type validation failed: encountered a pointer, but expected plain (non-pointer) bytes
   |
   = help: this indicates a bug in the program: it performed an invalid operation, and caused Undefined Behavior
   = help: see https://doc.rust-lang.org/nightly/reference/behavior-considered-undefined.html for further information
           
   = note: inside `main` at $DIR/transmute_fat1.rs:LL:CC

note: some details are omitted, run with `MIRIFLAGS=-Zmiri-backtrace=full` for a verbose backtrace

error: aborting due to previous error

